src/mon/Monitor.cc

```diff
@@ -2551,16 +2557,38 @@ void Monitor::log_health(
 	clog->warn() << ss.str();
       else
 	clog->error() << ss.str();
+
+      logged = true;
     } else {
       if (p.second.summary != q->second.summary ||
 	  p.second.severity != q->second.severity) {
-	// summary or severity changed (ignore detail changes at this level)
-	ostringstream ss;
+
+        auto status_iter = health_check_log_times.find(p.first);
+        if (status_iter != health_check_log_times.end()) {
+          if (now - status_iter->second.updated_at < min_log_period) {
+            // We already logged this recently, skip emitting an update.
+            // We'll get an update out of tick() later if the check
+            // is still failing.
+            continue;
+          }
+        }
+
+        // summary or severity changed (ignore detail changes at this level)
+        ostringstream ss;
         ss << "Health check update: " << p.second.summary << " (" << p.first << ")";
-	if (p.second.severity == HEALTH_WARN)
-	  clog->warn() << ss.str();
-	else
-	  clog->error() << ss.str();
+        if (p.second.severity == HEALTH_WARN)
+          clog->warn() << ss.str();
+        else
+          clog->error() << ss.str();
+
+        logged = true;
+      }
+
+      // Record the time at which we last logged, so that we can check this
```

Comment:
I believe Multi-line comments should come in /* ... */ as per coding guidelines.